ksql-common/src/test/java/io/confluent/ksql/schema/connect/SchemaWalkerTest.java

```diff
@@ -220,111 +224,140 @@ public void shouldVisitStruct() {
+    primitiveSchemas().forEach(schema -> {
 
       // When:
-      SchemaWalker.visit(schema, visitor);
+      final String result = SchemaWalker.visit(schema, visitor);
 
       // Then:
-      verify(visitor).visitInt(same(schema));
+      assertThat(result, is("Expected"));
     });
   }
 
   @Test
-  public void shouldVisitFloats() {
-    floatSchemas().forEach(schema -> {
+  public void shouldVisitAll() {
+    // Given:
+    visitor = new Visitor<String>() {
+      @Override
+      public String visitSchema(final Schema schema) {
+        return "Expected";
+      }
+    };
 
-      // Given:
-      clearInvocations(visitor);
+    allSchemas().forEach(schema -> {
 
       // When:
-      SchemaWalker.visit(schema, visitor);
+      final String result = SchemaWalker.visit(schema, visitor);
 
       // Then:
-      verify(visitor).visitFloat(same(schema));
+      assertThat(result, is("Expected"));
     });
   }
 
   @Test
-  public void shouldVisitNumbers() {
-    numberSchemas().forEach(schema -> {
+  public void shouldThrowByDefaultFromAll() {
+    // Given:
+    visitor = new Visitor<String>() {
+    };
 
-      // Given:
-      clearInvocations(visitor);
+    allSchemas().forEach(schema -> {
 
-      // When:
-      SchemaWalker.visit(schema, visitor);
+      try {
+        // When:
+        SchemaWalker.visit(schema, visitor);
 
-      // Then:
-      verify(visitor).visitNumber(same(schema));
+        fail();
+
+      } catch (final UnsupportedOperationException e) {
+        // Then:
+        assertThat(e.getMessage(), is("Unsupported schema type: " + schema));
+      }
     });
   }
 
-  public static Stream<Schema> integerSchemas() {
+  @Test
+  public void shouldThrowOnUnknownType() {
+    // Given:
+    final Type unknownType = mock(Type.class, "bob");
+    final Schema schema = mock(Schema.class);
+    when(schema.type()).thenReturn(unknownType);
+
+    // Then:
+    expectedException.expect(UnsupportedOperationException.class);
+    expectedException.expectMessage("Unsupported schema type: bob");
+
+    // When:
+    SchemaWalker.visit(schema, visitor);
+  }
+
+  public static Stream<Schema> primitiveSchemas() {
     return Stream.of(
+        Schema.OPTIONAL_BOOLEAN_SCHEMA,
         Schema.OPTIONAL_INT8_SCHEMA,
         Schema.OPTIONAL_INT16_SCHEMA,
         Schema.OPTIONAL_INT32_SCHEMA,
-        Schema.OPTIONAL_INT64_SCHEMA
-    );
-  }
-
-  public static Stream<Schema> floatSchemas() {
-    return Stream.of(
+        Schema.OPTIONAL_INT64_SCHEMA,
         Schema.OPTIONAL_FLOAT32_SCHEMA,
-        Schema.OPTIONAL_FLOAT64_SCHEMA
+        Schema.OPTIONAL_FLOAT64_SCHEMA,
+        Schema.OPTIONAL_STRING_SCHEMA
     );
   }
 
-  public static Stream<Schema> numberSchemas() {
+  @SuppressWarnings("UnstableApiUsage")
+  public static Stream<Schema> allSchemas() {
     return Streams.concat(
-        integerSchemas(),
-        floatSchemas()
+        primitiveSchemas(),
+        Stream.of(
+            Schema.OPTIONAL_BOOLEAN_SCHEMA,
```

Comment:
This looks like it should be adding non-primitive schemas instead.

Reply:
Yep - thanks. Old cut&paste error!